x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -463,9 +468,14 @@ export function registerIndexRoutes({
           inferenceConfig,
           client.asCurrentUser
         );
+        const mappingResponse = fieldMappings
+          ? (await updateMlInferenceMappings(indexName, fieldMappings, client.asCurrentUser))
```

Comment:
👍 I didn't want it there because `createAndReferenceMlInferencePipeline` doesn't imply mapping change side effects, but I didn't think about just renaming the function. Yep, that works for me.

Though, will you need any indication that the mapping change was successful separate from the pipeline creation being successful? Right now having them as two distinct function calls makes it easy to have the response indicate two distinct successes. Still doable as 1 function, just not sure if you want that or not.